### PR TITLE
Calendar view feature branch into RC

### PR DESCRIPTION
**Pull request by mdickson-adbe**
Calendar view for the program-details page and various fixes.

- Before: https://main--assets-distribution-portal--adobe.hlx.page/sample-public-site
- After: https://rc--adobe-gmo--hlxsites.hlx.page/program-details?programName=Summit%202025%20CSC%20Project&programID=66aa945b075a9e4cc04e149f1d72fb8c#

### CODE REVIEW

File: blocks/gmo-program-details/gmo-program-details.js [hunk 1]
```diff
@@ -177,6 +178,32 @@ export default async function decorate(block) {
...
+                <div class="inc-dec-wrapper">
+                    <div class="year-switch">
+                        <div id="dec-year" class="year-toggle">
+                            <img class="left" data-direction="left" src="/icons/chevron-right.svg"></img>
+                        </div>
+                        <div id="inc-year" class="year-toggle">
+                            <img class="right" data-direction="right" src="/icons/chevron-right.svg"></img>
+                        </div>
+                    </div>
+                    <div class="current-year" data-quarter="1" data-year="2024">2024</div>
```
Comment (shiv-gup) [inline blocks/gmo-program-details/gmo-program-details.js:192]: @mdickson-adbe can we get current year programatically?

Comment (mdickson-adbe) [reply blocks/gmo-program-details/gmo-program-details.js:192]: Sure, that's no problem at all.

**Review by shiv-gup — APPROVED** [review]

File: scripts/program-calendar.js [hunk 2]
```diff
@@ -0,0 +1,648 @@
...
+        const calendarTab = document.querySelector('.calendar.tab');
+        calendarTab.innerHTML = `
+            <div class="no-data-msg">Required Data is Unavailable for this view.</div>
+        `;
+        return;
+    }
+
+    // get start of the view
+    viewStart = getTimeBounds(deliverables, "start", startDateProp);
+    viewStart = (!(isValidDate(viewStart)) || viewStart.getFullYear() === 1969) ? programLaunchDate : viewStart;
```
[the remark this answers is not in the document]
Comment (mdickson-adbe) [reply scripts/program-calendar.js:42]: Good catch, it can be easy to forget other timezones on occasion.

[the remark this answers is not in the document]
Comment (mathieu-lessard) [reply scripts/program-calendar.js]: @mdickson-adbe thanks for making that change. There were a few other spots with 1969 being used as well, but I wasn't sure if I should make the same comment on all of them.

[the remark this answers is not in the document]
Comment (mdickson-adbe) [reply scripts/program-calendar.js]: I can check for any others. I have another bugfix I need to do for something unrelated and can put it into the same branch.

**Review by mathieu-lessard — APPROVED** [review]

**Pull request merged**